src/main/java/com/github/dockerjava/core/command/PullImageResultCallback.java

```diff
@@ -50,4 +50,25 @@ public void awaitSuccess() {
             throw new DockerClientException("Could not pull image: " + message);
         }
     }
+    
+    /**
+     * Awaits the image to be pulled successful, with timeout
+     *
+     * @throws DockerClientException
+     *             if the pull fails.
+     */
+    public void awaitSuccess(long timeout, TimeUnit timeUnit) {
+        try {
+            awaitCompletion(timeout, timeUnit);
+        } catch (InterruptedException e) {
+            throw new DockerClientException("", e);
+        }
+
+        if (latestItem == null) {
```

Comment:
@hazim1093 why do you duplicate this code? Extract it to separate method, see [BuildImageResultCallback](https://github.com/docker-java/docker-java/blob/master/src/main/java/com/github/dockerjava/core/command/BuildImageResultCallback.java) for reference